src/ipc/ipc4/handler.c

```diff
@@ -1680,6 +1680,7 @@ void ipc_cmd(struct ipc_cmd_hdr *_hdr)
 		}
 
 		/* copy contents of message received */
+		reply.primary.dat = 0;
```

Comment:
i feel like you should just initialize the structure or memset it, otherwise everytime we add a new field this is going to happen again.